apps/mitt-konto/src/MittKonto/Wrappers/ViewWrapper.purs

```diff
@@ -0,0 +1,164 @@
+  Basic -> viewWrapperBasic props
+  Async -> viewWrapperAsync props
+  where
+    viewWrapperBasic :: (Props p) -> JSX
+    viewWrapperBasic = make component
+      { initialState: initialStateBasic
+      , didMount: didMountBasic
+      , render: renderBasic
+      }
+
+    initialStateBasic =
+      { closeable: true
+      , closeAutomatically: Off
+      , titleText: mempty
+      , renderedContent: mempty
+      , onCancel: pure unit
+      }
+
+    didMountBasic :: React.Self (Props p) ViewWrapperStateBasic -> Effect Unit
+    didMountBasic self@{ props: { content }, setState } = do
+      instantiate content $ SetViewWrapperStateBasic setState
+
+    viewWrapperAsync :: (Props p) -> JSX
+    viewWrapperAsync = make component
+      { initialState: initialStateAsync
+      , didMount: didMountAsync
+      , render: renderAsync
+      }
+
+    initialStateAsync = Record.merge initialStateBasic
+      { asyncWrapperState: AsyncWrapper.Ready
+      , onTryAgain: pure unit
+      }
+
+    didMountAsync :: React.Self (Props p) ViewWrapperStateAsync -> Effect Unit
+    didMountAsync self@{ props: { content }, setState } = do
+      instantiate content $ SetViewWrapperStateAsync setState
+
+renderBasic :: forall p. (ViewWrapperContent p) => React.Self (Props p) ViewWrapperStateBasic -> JSX
+renderBasic self@{ props, state: { closeable, titleText, onCancel, renderedContent }, setState } =
+  render props self.state renderedContent
+
+renderAsync :: forall p. (ViewWrapperContent p) => React.Self (Props p) ViewWrapperStateAsync -> JSX
+renderAsync self@{ props: { content, closeType, wrapperType }, state: { closeable, closeAutomatically, titleText, onCancel, onTryAgain, renderedContent }, setState } =
+  render
+    self.props
+    { closeable
+    , closeAutomatically
+    , titleText
+    , onCancel
+    , renderedContent
+    }
+    asyncWrapper
+  where
+    asyncWrapper :: JSX
+    asyncWrapper = AsyncWrapper.asyncWrapper
+      { wrapperState: self.state.asyncWrapperState
+      , readyView: renderedContent
+      , editingView: identity
+      , successView: \msg -> successWrapper msg
+      , errorView: \err -> errorWrapper onTryAgain err
+      , loadingView: identity
+      }
+
+render :: forall p. (ViewWrapperContent p) => Props p -> ViewWrapperStateBasic -> JSX -> JSX
+render props@{ route, routeFrom } state@{ closeable, closeAutomatically, titleText, onCancel, renderedContent } content =
+  Router.route
+    { exact: true
+    , path: Just route
+    , render: \_ -> DOM.div_
+        [ header
+        , content
+        ] <> case closeAutomatically of
+               On delay -> autoClose props delay
+               Off      -> mempty
+    }
+  where
+    header :: JSX
+    header = Grid.row_
+      [ DOM.div
+          { className: "col col-11"
+          , children: [ title ]
+          }
+      , if closeable then
+          DOM.div
+            { className: "col-1 flex credit-card-choice--close-icon"
```

Comment:
I take it that `ViewWrapper.purs` is meant for other users besides credit cards? This class name wouldn't belong here then.

Reply:
Yes of course that is not right 😃